### PR TITLE
feat: add Toast component

### LINKED ISSUE #264: Growl container

### PR DESCRIPTION
**Pull request by theo-mesnil**
Fixes #264 #510

### CODE REVIEW

File: packages/Toast/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# @welcome-ui/toast v1.0.0
+
+The [Growl](http://welcome-ui.com/components/toast) component from [@welcome-ui](http://welcome-ui.com).
```
Comment (anucreative) [inline packages/Toast/README.md]:
```suggestion
The [Toast](http://welcome-ui.com/components/toast) component from [@welcome-ui](http://welcome-ui.com).
```

File: packages/Toast/doc.mdx [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+      >
+        Show Toast
+      </Button>
+    )
+  }}
+</Playground>
+
+## Options
+
+You can choose between 2 options :
```
Comment (anucreative) [inline packages/Toast/doc.mdx]:
```suggestion
The `toast` function takes a component to be displayed and an `options` object:
```

File: packages/Toast/index.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import { ThemeContext, ThemeProvider } from '@xstyled/styled-components'
+import toast from 'toasted-notes'
+
+import * as S from './styles'
+
+export function useToast() {
+  const themeContext = useContext(ThemeContext)
+
+  function createToast(children, options) {
+    const getOptions = {
```
Comment (anucreative) [inline packages/Toast/index.js]:
```suggestion
    const toastOptions = {
```

(Ce n'est pas un fonctionne donc on a pas besoin de `getXxx`)

File: packages/Toast/package.json [hunk 4]
```diff
@@ -34,20 +34,23 @@
...
-    "@xstyled/styled-components": "^1.15.1",
-    "@xstyled/system": "^1.15.1",
-    "prop-types": "^15.7.2",
+    "@xstyled/styled-components": "^1.15.0",
+    "@xstyled/system": "^1.14.0",
     "react": "^16.10.2",
     "react-dom": "^16.10.2",
-    "styled-components": "^4.0.0 || ^5.0.0"
+    "styled-components": "^4.0.0 || ^5.0.0",
+    "prop-types": "^15.7.2"
```
Comment (anucreative) [inline packages/Toast/package.json]: Alphabetical order. Normalement lint le corrige, non ?

File: packages/Toast/package.json [hunk 5]
```diff
@@ -34,20 +34,23 @@
...
   },
   "gitHead": "5e05d638c444651c818138903aef4a68cbbb5747",
   "sideEffects": false,
-  "component": "Growl",
-  "documentation": "http://welcome-ui.com/components/growl"
-}
\ No newline at end of file
+  "component": "Toast"
```
Comment (anucreative) [inline packages/Toast/package.json]:
```suggestion
  "component": "Toast",
  "documentation": "http://welcome-ui.com/components/toast"
```

File: src/utils/propTypes.js [hunk 6]
```diff
@@ -24,4 +24,4 @@ export const SHAPES_TYPE = ['square', 'circle']
 
 export const SIZES_TYPE = ['sm', 'md', 'lg']
 
-export const VARIANTS_TYPE = ['error', 'info', 'valid', 'warning']
+export const VARIANTS_TYPE = ['error', 'info', 'valid', 'warning', 'success']
```
Comment (anucreative) [inline src/utils/propTypes.js]: Alphabetical order

File: packages/Toast/doc.mdx [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+            {position}
+          </Button>
+        ))}
+      </>
+    )
+}}
+
+</Playground>
+
+## Snackbar
```
Comment (anucreative) [inline packages/Toast/doc.mdx]: ## Supplied components

You can create your own notification component to display in the Toast or one of our default components: `Snackbar` and `Growl`. 

In general…
- a `Snackbar` is used to display a temporary, information-only notification, and 
- a `Growl` is used to display a more important notification with an action (that requires the user to close the notification)

### Snackbar

…

File: packages/Toast/Growl/Title.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
+import React from 'react'
+import { node, oneOf } from 'prop-types'
+
+import { VARIANTS_TYPE } from '../../../src/utils/propTypes'
+
+import { Title as StyledTitle } from './styles'
```
Comment (P1X3L) [inline packages/Toast/Growl/Title.js]:
```suggestion
import * as S from './styles'
```

File: packages/Toast/Growl/Title.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+
+export const Title = ({ children, dataTestId, variant = 'info' }) => (
+  <StyledTitle data-testid={dataTestId} variant={variant}>
```
Comment (P1X3L) [inline packages/Toast/Growl/Title.js]:
```suggestion
  <S.Title data-testid={dataTestId} variant={variant}>
```

File: packages/Toast/Growl/Title.js [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+    {children}
+  </StyledTitle>
```
Comment (P1X3L) [inline packages/Toast/Growl/Title.js]:
```suggestion
  </S.Title>
```

File: packages/Toast/package.json (hunk 4 above)
Comment (theo-mesnil) [reply packages/Toast/package.json]: not on package json i think

File: packages/Toast/Growl/index.js [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+import { Title } from './Title'
+
+export const Growl = forwardRef(({ children, dataTestId, onClose }, ref) => (
+  <S.Growl ref={ref}>
+    {onClose && (
+      <ClearButton
+        data-testid={dataTestId}
+        onClick={onClose}
+        position="absolute"
+        right={10}
```
Comment (anucreative) [inline packages/Toast/Growl/index.js]: On peut utiliser le theme la e.g. `right="sm"` ?

Comment (theo-mesnil) [reply packages/Toast/Growl/index.js]: Non cela ne marche pas avec xstyled system... :'(

File: packages/Toast/package.json [hunk 12]
```diff
@@ -34,20 +34,24 @@
...
     "@welcome-ui/system": "^1.5.2",
-    "@welcome-ui/utils": "^1.4.0"
+    "@welcome-ui/utils": "1.4.0",
+    "react-spring": "^8.0.27",
+    "toasted-notes": "^3.2.0"
   },
   "peerDependencies": {
-    "@xstyled/styled-components": "^1.15.1",
-    "@xstyled/system": "^1.15.1",
+    "@xstyled/styled-components": "^1.15.0",
```
Comment (P1X3L) [inline packages/Toast/package.json]: pourquoi passer à `1.15.0` ???

File: packages/Toast/package.json [hunk 13]
```diff
@@ -34,20 +34,24 @@
...
+    "@xstyled/system": "^1.14.0",
```
Comment (P1X3L) [inline packages/Toast/package.json]: pareil içi ??

File: packages/Toast/Growl/index.js (hunk 11 above)
Comment (P1X3L) [reply packages/Toast/Growl/index.js]: https://github.com/WTTJ/welcome-ui/blob/feat-toast/packages/ClearButton/styles.js#L17
Le ClearButton n'utilise pas xstyled, il suffit de lui passer les system props, cf --> https://github.com/WTTJ/welcome-ui/blob/5cacf8100d0d682f0b38378328175be54bc17a04/packages/Card/styled.js#L3

Comment (theo-mesnil) [reply packages/Toast/Growl/index.js]: bug côté xstyled > https://github.com/smooth-code/xstyled/issues/92

**Pull request merged**